nixos/modules/services/security/nwaku.nix

```diff
@@ -0,0 +1,318 @@
+{ config, lib, pkgs, ... }:
+
+let
+  inherit (lib)
+    types mkEnableOption mkOption mkIf forEach
+    escapeShellArg mdDoc literalExpression
+    optionals concatStringsSep stringLength boolToString;
+
+  cfg = config.services.nwaku;
+in {
+  options = {
+    services = {
+      nwaku = {
+        enable = mkEnableOption (mdDoc "Nim Waku node service.");
+
+        package = mkOption {
+          type = types.package;
+          default = pkgs.nwaku;
+          defaultText = literalExpression "pkgs.nwaku";
+          description = mdDoc "Package to use as Nim Waku node.";
+        };
+
+        protocols = mkOption {
+          type = types.listOf types.str;
+          default = ["relay" "filter" "swap" "lightpush"];
+          example = ["relay" "store" "filter" "swap" "lightpush" "rln-relay" "relay-peer-exchange"];
+          description = mdDoc "List of protocols to enable on the node.";
+        };
+
+        nodekey = mkOption {
+          type = types.nullOr types.str;
+          default = null;
+          example = "abcd1234abcd1234abcd1234abcd1234abcd1234abcd1234abcd1234abcd1234";
+          description = mdDoc "P2P node private key as 64 char hex string.";
+        };
```

Comment:
Good point.

Reply:
Is there a way to pass it the value from a file to a CLI flag in `systemd` service definition if the node can't read node key from a file?

Reply:
Actually, maybe we can work around it using env vars:
https://github.com/waku-org/nwaku/blob/master/docs/operators/how-to/configure.md#environment-variables

Reply:
There we go:
https://github.com/NixOS/nixpkgs/blob/66fee4e79f0c1734b5f340cc9c11c89e580d87cd/nixos/modules/services/security/nwaku.nix#L35-L43
https://github.com/NixOS/nixpkgs/blob/66fee4e79f0c1734b5f340cc9c11c89e580d87cd/nixos/modules/services/security/nwaku.nix#L269
Not perfect, since if the env var name is mistyped the node will keep going and just generate a random key.
But we can fix that once the next release includes a proper flag for node key file.